cloud/scope/powervs_machine.go

```diff
@@ -577,12 +587,38 @@ func (m *PowerVSMachineScope) createCOSClient() (*cos.Service, error) {
 		fmt.Printf("ibmcloud api key is not provided, set %s environmental variable", "IBMCLOUD_API_KEY")
 	}
 
-	cosClient, err := cos.NewService(cos.ServiceOptions{}, m.IBMPowerVSCluster.Spec.CosInstance.BucketRegion, apiKey, *serviceInstance.GUID)
+	region := m.IBMPowerVSCluster.Spec.CosInstance.BucketRegion
+	// if the bucket region is not set, use vpc region
+	if region == "" {
+		vpcDetails := m.IBMPowerVSCluster.Spec.VPC
+		if vpcDetails == nil || vpcDetails.Region == nil {
+			return nil, fmt.Errorf("failed to determine cos bucket region, both buckeet region and vpc region not set")
+		}
+		region = *vpcDetails.Region
+	}
+
+	serviceEndpoint := fmt.Sprintf("s3.%s.%s", region, cosURLDomain)
+	// Fetch the COS service endpoint.
+	cosSvcEndpoint := endpoints.FetchEndpoints(string(endpoints.COS), m.ServiceEndpoint)
```

Comment:
can we rename `cosSvcEndpoint` to something like - `cosEndpoint` or `cosServiceEndpoint`

Reply:
Renamed as suggested.